src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -481,6 +482,9 @@ private void processPackage(DetailAST ast, DetailAST nextToken) {
         if (!hasEmptyLineAfter(ast)) {
             log(nextToken.getLineNo(), MSG_SHOULD_BE_SEPARATED, nextToken.getText());
         }
+        findFirstModifier(nextToken)
+            .filter(mod -> mod.getLineNo() == ast.getLineNo() + 1)
+            .ifPresent(mod -> log(mod.getLineNo(), MSG_SHOULD_BE_SEPARATED, mod.getText()));
```

Comment:
probably it is good time to start discussion and agree on how far we go with functional style.
In general I am in ok, but here is special case of `log` in lambda.
In most cases I demand `log` to be placed in `visitToken` with some exceptions when Check is complicated, or it was like this for long time, and we do not have time for refactor.
I think that we should not place `log` into lambda, because we lost ability of coverage/pitest to automatically show us such line is covered. Functional style is very good style when engineering quality if measured by coverage, single test can cover all code :), but this is not a reason why we use coverage. Coverage helps us to grow Input code base with cases. Contributors always try to provide as minimum tests as possible.

If would do:
```
if (variable.isPresent()) {
  log()
}
``` 

@rnveach , @pbludov , please vote of your preferences.

Reply:
> I think that we should not place `log` into lambda, because we lost ability of coverage/pitest to automatically show us such line is covered

I asked this question some time ago, googled a little and tried it myself. And I haven't seen any problems with pitest and lambdas.
Can you share an example where lambda hides coverage?

Reply:
If something hurts pitest or coverage, I too would like to avoid it. If we know something will break pitest, we should write a check to avoid it.
@strkkk Can you post pitest report for your changes?

My own personal opinion is I don't like Java 8+ functionality anyways. I would much rather see everything written it out as I don't find lambdas very clear to read.

Reply:
pitest and jacoco coverage will be green and 100% covered.
pitest does not have such mutations to substitude lambda to somethig else.
jacoco it is not smart enough for now to detect actual lambda body execution.
BUT it only mean that particular line was executed by some tests. Pitest will not do any mutation as far as I remember, so it will be green.

Functional style code is very easy to cover, as all IFs (`ifPresent`) are chained in single statement, and jacoco consider them as single statement. Any test will cover it.
Well known cheating of pitest mutations is to substitute all "+-" with `add/remove` methods. Amount of mutations will be seriously reduced, so easy to pass.

Java 8+ functionality is ok, it ease code in some cases, but only if we know that do coverage of all(very most) branches it imply.
Fight for coverage, removed a lot of nonsense code and heavily forced us to extend Input code base. Idea of almost complete ban of pure UTs is also come from idea that we do not care about coverage number, we want to have Inputs that cover code and all cases (functional test).

Our CI will force contributor to provide heavy testing if code is old-styled-java(imperative), it was good time for maintainers during code review as you now that each line/branch was covered by Input. 

So even we make our code modern in some parts, I would rather keep at least `logs` calls out of such single-long statement(long chaining of calls). I am a bit paranoid probably.

@pbludov , please share your ideas on this topic.

Reply:
> My own personal opinion is I don't like Java 8+ functionality anyways. I would much rather see everything written it out as I don't find lambdas very clear to read

Not about lambdas, but optional style is a best way to dig down 2+ nullable levels down in  AST, because it eliminates useless null checks.

> Pitest will not do any mutation as far as I remember, so it will be green.

Pitest mutates conditions, even in lambdas.
Ok, I can replace ifPresent with explicit `isPresent` call

Reply:
> Ok, I can replace ifPresent with explicit isPresent call

Yes, it will shift the focus in the right direction.

Reply:
changed.

Now Idea complains that this can be changed to use functional style :)

Reply:
you already should know that static analysis is full of false-positives and exceptions of rules. It is better to suppress violation if we know a good reason.